generate_network.py

```diff
@@ -90,28 +92,55 @@ def get_random_sublist(topics):
         sublist.append(topics[i])
     return sublist
 
-def generate_network(num_nodes, prefix):
+
+# Network Types
+# https://networkx.org/documentation/stable/reference/generated/networkx.generators.degree_seq.configuration_model.html
+def generate_config_model(n):
+    #degrees = nx.random_powerlaw_tree_sequence(n, tries=10000)
+    degrees = [random.randint(1, n) for i in range(n)]
+    if (sum(degrees)) % 2 != 0:         # adjust the degree to even
+        degrees[-1] += 1
+    return nx.configuration_model(degrees) # generate the graph
+
+
+def generate_scalefree_graph(n):
+    return  nx.scale_free_graph(n)
+
+
+# n must be larger than k
+def generate_newman_watts_strogatz_graph(n):
+    return nx.newman_watts_strogatz_graph(n, 12, 0.5)
+
+
+# Generate the network from nw type
+def generate_network(num_nodes, nw_type, prefix):
     G = nx.empty_graph()
     if nw_type == "configuration_model":
         G = generate_config_model(num_nodes)
+    elif nw_type == "scalefree":
+        G = generate_scalefree_graph(num_nodes)
+    elif nw_type == "newman_watts_strogatz":
+        G = generate_newman_watts_strogatz_graph(num_nodes) 
     else: 
         print(nw_type +": Unsupported network type")
         sys.exit(1)
     H = postprocess_network(G, prefix)
     return H
```

Comment:
Nitpick:

You can use a **dictionary dispatch** here.  Labels for selecting operations shouldnt be strings but proper types (probably an enum). That way you can pop up the error much earlier in the computation when reading the `nw_type` from the configuration. Otherwise it should never break.

Reply:
True, but the network types as ordinals/enums is a tad un-intuitive. Also, this will require the script to list (in help) the supported enums/types and their  names. enums (and their associated 'bounds' checking) definitely is the way to go for interactive scripts. so let me put it in the todo :)